admin/ajax.php

```diff
@@ -54,16 +54,41 @@ function wpseo_set_ignore() {
 
 	check_ajax_referer( 'wpseo-ignore' );
 
-	$options                            = get_option( 'wpseo' );
-	$ignore_key                         = sanitize_text_field( WPSEO_Utils::filter_input( INPUT_POST, 'option' ) );
-	$options[ 'ignore_' . $ignore_key ] = true;
-	update_option( 'wpseo', $options );
+	$ignore_key = sanitize_text_field( WPSEO_Utils::filter_input( INPUT_POST, 'option' ) );
```

Comment:
It might be better to call `filter_input` directly instead of calling the wrapper.